File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int>mp;
        vector<vector<int>> ans;
        vector<int> res;

        for(int n:nums){
            mp[n]++;
        }

        for(auto it: mp){
            ans.push_back({it.second,it.first});

        }

        sort(ans.begin(),ans.end(), [](const vector<int>& a, const vector<int>& b){
            return a[0]>b[0];
        });

        for(int i=0; i<k; i++){
            res.push_back(ans[i][1]);
        }
        return res;
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        prefix = 0
        count = 0

        mp = {0:1}

        for n in nums:

            prefix += n

            if prefix - k in mp:
                count += mp[prefix - k]
            
            mp[prefix] = mp.get(prefix,0)+1
        return count
    